lib/discordrb/data/message.rb

```diff
@@ -390,5 +357,36 @@ def buttons
 
       results.flatten.compact
     end
+
+    # The user who sent this message message. Will be a {Member} most of the time, a {User} in the case
+    # that the user has left the server, or a {Recipient} if the message was sent in a private channel.
+    def author
+      author_id = @author_data['id'].to_i
+      return unless author_id
+
+      if @author_data['discriminator'] == ZERO_DISCRIM
+        # This is a webhook user! It would be pointless to try to resolve a member here, so we just create
+        # a User and return that instead.
+        Discordrb::LOGGER.debug("Webhook user: #{author_id}")
+        User.new(data['author'], @bot)
+      elsif @channel.private?
+        # Turn the message user into a recipient - we can't use the channel recipient
+        # directly because the bot may also send messages to the channel
+        Recipient.new(@bot.user(author_id), @channel, @bot)
+      else
+        author = @channel.server.member(author_id) || @bot.user(author_id)
+        case author
+        when Member
+          author.update_data(@author_data)
```

Comment:
Because we're handling this lazily now we should not update at this point in the lifecycle. There is no guarantee that our data is more up to date than the cache

Reply:
We may need to ensure that we're updating/creating the member during the MessageCreate event processing though

Reply:
Is this what `ensure_user` is for?